src/shared/proc.cpy

```diff
@@ -110,13 +113,16 @@ namespace proc:
       base = basename((char *) base)
       base_str := string(base)
 
+      auto search_interpreter = known_interpreters.find(base_str)
+      if search_interpreter != known_interpreters.end():
+          continue
```

Comment:
the logic here is a bit unclear to me. what happens if: "python -n foo.py" is a cmdline and args is { "foo.py" }? 

i think this is supposed to only check the rest of cmd_tokens if the first token is the interpreter, right? (but it should check all of the cmdline, not just the second token?)

Reply:
Yeah, my main idea was that we just to check the first argument of the cmdline to get the name of the running program. But then I thought about cases like shell or python scripts where the check if `bash` or `python` is running is not really meaningful. Therefore the approach was to skip the first token and check until the first token without an leading `-`. I know this is still buggy in corner cases when the cmdline is something like `python -X <some path> <command>` but I had no quick idea how to catch these cases in general.

Reply:
sorry for the delay, i will revisit this PR this week (and next) and try to fix things (and write some tests:D). for args with '-X' foo, i would just skip over any args that start with '-' until we get to the first one